canvas/templatetags/event_grade.py

```diff
@@ -0,0 +1,24 @@
+from functools import reduce
+from django import template
+from django.db.models import Sum, F, Count
+from course.utils.utils import get_token_value
+
+register = template.Library()
+
+
+@register.filter
+def total_event_grade(event, user):
```

Comment:
It's better not to have this much logic in the template tags. What I would recommend is to create a function in utils to calculate the total event grade and call it here in the template tag.